### PR TITLE
Modified Interview and Company APIs

### PR DESCRIPTION
**Pull request by shahparth36**
Added cache and login middleware to interview and company apis.
Added userId to interview model

### CODE REVIEW

File: server/src/controllers/CompanyController.js [hunk 1]
```diff
@@ -42,22 +42,26 @@ const createCompany = async (req, res) => {
     }
 };
 
-const getCompanies = async (req, res) => {
+const getCompanies = async (req, res, next) => {
     try {
         const companies = await Company.find();
+        // res.locals.cache = companies;
+        // next();
```
Comment (ShubhankarKG) [inline server/src/controllers/CompanyController.js:49]: Please remove comments if not needed. Also, if next is not needed we can remove it.

File: server/src/controllers/InterviewController.js [hunk 2]
```diff
@@ -52,23 +53,27 @@ const getInterviewByCompanyID = async (req, res) => {
...
+const getInterviewByID = async (req, res, next) => {
     try {
         // if (!req.query || !req.query.interviewTitle) {
         //     return res.status(422).json({ error: 'Please provide interview title' });
         // }
         const interviewID = req.params.id;
         // const interview = await Interview.findOne({ _id: interviewID }).populate('company').exec();
         const interview = await Interview.findOne({ _id: interviewID });
+        // res.locals.cache = interview;
+        // next();
```
Comment (ShubhankarKG) [inline server/src/controllers/InterviewController.js:65]: Please remove the comments and the next parameter if not needed.

**Review by ShubhankarKG — CHANGES_REQUESTED** [review]

**Comment by shahparth36** [comment]
Have removed comments from controller as requested

**Review by ShubhankarKG — APPROVED** [review]

**Comment by RaviMauryaHootowl** [comment]
Thanks!

**Pull request merged**